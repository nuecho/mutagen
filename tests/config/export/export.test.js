test(`[mutagen config export --help] should print import usage`, () => {
  assertMutagenResult(`config export --help`, "export usage", 0);
});

test(`[mutagen config export] should throw an exception when the encoding specified is not supported`, () => {
  const { code, output } = mutagen(`-e invalidEncoding config export`);

  expect(output.stderr).toMatchSnapshot("export invalid encoding exception");
  expect(code).toBe(1);
});

test(`[mutagen config export] should not throw an exception when no encoding is specified`, () => {
  const { code, output } = mutagen(`config export`);

  expect(output.stderr).toMatchSnapshot("export without exception");
  expect(code).toBe(0);
});

test(`[mutagen config export] should export the config in raw format`, () => {
     const { code, output } = mutagen(`config export`);
     const exportedConfig = JSON.parse(output.stdout);

     expect(exportedConfig.CFGEnumerator).toBeDefined;
     expect(exportedConfig.CFGPerson).toBeDefined;
     expect(exportedConfig.CFGTenant).toBeDefined;
     expect(code).toBe(0);
   });

test(`[mutagen config export --format JSON] should export the config in json format`, () => {
     const { code, output } = mutagen(`config export --format JSON`);
     const exportedConfig = JSON.parse(output.stdout);

     expect(exportedConfig.enumerators).toBeDefined;
     expect(exportedConfig.persons).toBeDefined;
     expect(exportedConfig.tenants).toBeDefined;
     expect(code).toBe(0);
   });