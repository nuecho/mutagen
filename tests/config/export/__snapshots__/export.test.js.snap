// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[mutagen -p config export] should allow passing password on standard input: export stdin password 1`] = `""`;

exports[`[mutagen config export --help] should print import usage: export usage 1`] = `
Object {
  "stderr": "",
  "stdout": "Usage: export [-?] [--trust-insecure-certificate] [--format=<format>]
Export configuration objects.
      --format=<format>       Export format [RAW|JSON|COMPACT_JSON].
      --trust-insecure-certificate
                              Don't validate the server's TLS certificate.
  -?, -h, --help              Show this help message.
",
}
`;

exports[`[mutagen config export] should not throw an exception when no encoding is specified: export without exception 1`] = `""`;

exports[`[mutagen config export] should throw an exception when the encoding specified is not supported: export invalid encoding exception 1`] = `
"Encoding 'test' specified in environments.yml is not supported by confServer connection.
"
`;
